packages/core/src/provider/base.ts

```diff
@@ -540,4 +562,33 @@ export class SmartAccountProvider<
       };
     };
   };
+
+  private _runMiddlewareStack = async (
```

Comment:
We should use this not just for drop and replace but also in the main function too (_buildUserOp)

Reply:
For this, I will have to create a utility `UserOperationCalldata` to `UserOperationStruct`.

I am thinking of keeping it in the `base.ts` file itself since it is specific to provider and not a general utility function.